src/ngraph/op/fused/space_to_depth.cpp

```diff
@@ -43,57 +44,89 @@ op::SpaceToDepth::SpaceToDepth(const Output<Node>& data, const std::string& mode
 NodeVector op::SpaceToDepth::decompose_op() const
 {
     auto data = input_value(0);
-    const Shape& data_shape = data.get_shape();
+    auto data_shape = data.get_shape();
 
-    // Set default values to each dimension to be able to work with both 3D or 4D data.
-    size_t n{1}, c{1}, h{1}, w{1};
+    NGRAPH_CHECK((data_shape.size() >= 3),
+                 "The input tensor with rank lower than 3 is not supported (input rank: ",
+                 data_shape.size(),
+                 ")");
 
-    NGRAPH_CHECK((data_shape.size() == 3 || data_shape.size() == 4),
-                 "The provided tensor shape: ",
-                 data_shape,
-                 " is not supported.");
+    NGRAPH_CHECK(m_blocksize > 0, "m_blocksize must be greater than 0");
 
-    // Assume NCHW data layout
-    if (data_shape.size() == 4)
+    if (data_shape.size() == 3)
     {
-        n = data_shape.at(0);
-        c = data_shape.at(1);
-        h = data_shape.at(2);
-        w = data_shape.at(3);
-    }
-    // Without batch.
-    else if (data_shape.size() == 3)
-    {
-        c = data_shape.at(0);
-        h = data_shape.at(1);
-        w = data_shape.at(2);
+        // Insert batch axis
+        data_shape.insert(data_shape.begin(), 1);
+        data = builder::reshape(data, data_shape);
     }
 
-    NGRAPH_CHECK((h % m_blocksize == 0 && w % m_blocksize == 0 && m_blocksize > 0),
-                 "SpaceToDepth: The width and height axes size must be a multiple of ",
-                 "squared block_size attribute value");
+    const size_t n_dim = data_shape.at(0);
+    const size_t c_dim = data_shape.at(1);
+    const size_t no_spatial_dims = 2;
```

Comment:
Could you rename it to spatial_dims_index or something like this?

Reply:
I use `spatial_dim_index ` both for `DepthToSpace` and `SpaceToDepth`